src/traversal.js

```diff
@@ -16,11 +16,25 @@
 
 var getWalker = require('./walker').getWalker;
 var getData = require('./node_data').getData;
+var namespace = require('./namespace');
+
+
+/**
+ * Enters a Element, setting the current namespace for nested elements.
+ * @param {!Element} node
+ */
+var enterNode = function(node) {
+  var data = getData(node);
+
+  markVisited(node);
```

Comment:
It doesn't seem like we should need to call mark visited here, just on nextSibling